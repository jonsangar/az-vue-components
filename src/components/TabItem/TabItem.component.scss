.az-tabs {
  .az-tab {
    width: 100%;
    // padding: half() 0;
    font-weight: font-weight(bold);
    text-align: center;

    a{
      display: inline-block;
      width: 100%;
      padding: half() 0;
    }

    &:not(:last-child) {
      border-right: 1px solid color(border);
    }

    &--selected{
      color: white;
      text-shadow: 0 2px 4px rbga(var(--color-shadow), 0.08);
      background-image: radial-gradient(circle at 0% 0%, color(brand-light), color(brand-dark));
    }
  }
}
