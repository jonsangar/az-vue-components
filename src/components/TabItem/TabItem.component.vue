<template lang="pug">
  .az-tab(:class="{'az-tab--selected':isActive}")
    router-link(v-if="link !== ''" :to="{name: link}")
      slot

</template>

<script>
export default {

name: 'az-tab',

props: {

  isActive: {
    type: Boolean,
    default: false
  },

  link: {
    type: String,
    default: ''
  }

}

}
</script>

<style src="./TabItem.component.scss" lang="scss" scoped></style>
