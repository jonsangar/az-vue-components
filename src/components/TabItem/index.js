import TabItem from './TabItem.component.vue';

export default TabItem;
