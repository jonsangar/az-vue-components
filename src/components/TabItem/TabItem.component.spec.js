import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerías de estilos e iconos
import SvgIcon from 'vue-svgicon'
import TabItem from './TabItem.component.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('TabItem.vue', () => {

  const build = () => {
    const wrapper = shallowMount(TabItem, {
      localVue,
      propsData: {
        isActive: true,
        link: 'airtools',
      },
      stubs: ['router-link']
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente TabItem con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Comprueba que se actualizan las props cuando se asignan las props', async () => {
    const { wrapper } = build()

    expect(wrapper.vm.isActive).toBe(true)
    expect(wrapper.vm.link).toBe('airtools')
  });

  it('Comprueba que si está la pestaña activa se renderiza la clase az-tab--selected', async () => {
    const { wrapper } = build()

    expect(wrapper.find('.az-tab--selected').exists()).toBe(true)
  });

});
