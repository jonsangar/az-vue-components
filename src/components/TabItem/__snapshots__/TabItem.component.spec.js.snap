// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabItem.vue Comprueba renderizado del componente TabItem con snapshot 1`] = `
<div class="az-tab az-tab--selected">
  <router-link-stub to="[object Object]"></router-link-stub>
</div>
`;
